torch/csrc/autograd/engine.cpp

```diff
@@ -154,8 +154,14 @@ struct ReadyQueue {
 
 int NodeTask::getReentrantDepth() const {
   std::shared_ptr<GraphTask> graph_task = base_.lock();
-  TORCH_INTERNAL_ASSERT(graph_task, "GraphTask is no longer valid!")
-  return graph_task->reentrant_depth_;
+  if (graph_task) {
+    return graph_task->reentrant_depth_;
+  } else {
+    // The graph task is no longer valid indicating an error. As a result, we
+    // try to move this to the front of the queue to ensure the autograd
+    // engine threads pick up this error soon.
+    return std::numeric_limits<int>::max();
```

Comment:
nit: Maybe we want to introduce some different levels of "max" here. I think this soon be higher priority than the AccumulateGrad [here](https://github.com/pytorch/pytorch/blob/6e38a709397ba2299e99011618f78dd5ae4c048a/torch/csrc/autograd/functions/accumulate_grad.cpp#L20) that use `UINT64_MAX`.

Reply:
Doesn't AccumulateGrad use sequence_nr, but this is actually using reentrant_depth? Looking at CompareNodeTaskTime, it looks like we look at sequence_nr only if reentrant_depth is same. So this will be higher priority than AccumulateGrad.

I think we can add more fine grained levels when we have another case similar to this.

Reply:
Ho right. Sounds good then.
This is a bit hacky but that will work.